### PR TITLE
Bug 1904538: Dockerfile.ocp: specify numeric uid

### PR DESCRIPTION
**Pull request by s-urbaniak**
/cc @openshift/openshift-team-monitoring

### CODE REVIEW

File: Dockerfile.ocp [hunk 1]
```diff
@@ -17,6 +17,6 @@ LABEL io.k8s.display-name="kube-rbac-proxy" \
 ARG FROM_DIRECTORY=/go/src/github.com/brancz/kube-rbac-proxy
 COPY --from=builder ${FROM_DIRECTORY}/_output/kube-rbac-proxy  /usr/bin/kube-rbac-proxy
 
-USER nobody
+USER 65534
```
Comment (lilic) [inline Dockerfile.ocp:20]: So we just want this to be nobody but numericly? 

In CMO we use another user UID, hence the question.

**Review by lilic — COMMENTED** [review]
/lgtm

/hold

One question feel free to remove hold, just curious

File: Dockerfile.ocp (hunk 1 above)
Comment (s-urbaniak) [reply Dockerfile.ocp:20]: correct, we still want this to be nobody, but numeric. the whole problem is that if we use the literal string `"nobody"` non-numerically, along with the `nonroot` security context constraint, the provisioning will fail with the following error:
```
container has runAsNonRoot and image has non-numeric user (nobody), cannot verify user is non-root
```

Comment (s-urbaniak) [reply Dockerfile.ocp:20]: and we tried removing the `nonroot` security context constraint from CMO alltogether, but that didn't work of the following provisioning error: https://github.com/openshift/cluster-monitoring-operator/pull/1015#issuecomment-742416683

Comment (lilic) [reply Dockerfile.ocp:20]: Thanks for the explanation!!

/hold cancel

Comment (s-urbaniak) [reply Dockerfile.ocp:20]: side note: i am not very happen about the solution to be honest, the question around the mystery why cluster-monitoring-operator _sometimes_ gets `nonroot` assigned vs. `restricted` still remains (hypothesis so far is some race in the provisioning controller). Additionally, i do get the necessity for cluster-monitoring-operator needs to have the same security context constraint, but it is not really necessary for the pod runtime as we don't mount things there.

**Comment by s-urbaniak** [comment]
test failure seems unrelated, CMO and kube-rbac-proxy started successfully

**Pull request merged**